src/highdicom/legacy/sop.py

```diff
@@ -417,66 +545,1911 @@ def __init__(
-            )
-        if ref_ds.SOPClassUID != '1.2.840.10008.5.1.4.1.1.4':
-            raise ValueError(
-                'Wrong SOP class for conversion of legacy MR images.'
-            )
-
         sop_class_uid = LEGACY_ENHANCED_SOP_CLASS_UID_MAP[ref_ds.SOPClassUID]
-
+        all_framesets = FrameSetCollection(legacy_datasets)
+        if len(all_framesets.frame_sets) > 1:
+            raise ValueError(
+                'Mixed frames sets: the input single frame list contain more '
+                'than one multiframe collection')
+        frame_set = all_framesets.frame_sets[0]
+        if sort_key is None:
+            sort_key = _CommonLegacyConvertedEnhanceImage.default_sort_key
         super().__init__(
             study_instance_uid=ref_ds.StudyInstanceUID,
             series_instance_uid=series_instance_uid,
             series_number=series_number,
             sop_instance_uid=sop_instance_uid,
             sop_class_uid=sop_class_uid,
             instance_number=instance_number,
-            manufacturer=ref_ds.Manufacturer,
-            modality=ref_ds.Modality,
-            transfer_syntax_uid=None,  # FIXME: frame encoding
-            patient_id=ref_ds.PatientID,
-            patient_name=ref_ds.PatientName,
-            patient_birth_date=ref_ds.PatientBirthDate,
-            patient_sex=ref_ds.PatientSex,
-            accession_number=ref_ds.AccessionNumber,
-            study_id=ref_ds.StudyID,
-            study_date=ref_ds.StudyDate,
-            study_time=ref_ds.StudyTime,
-            referring_physician_name=ref_ds.ReferringPhysicianName,
-            **kwargs
+            manufacturer="" if 'Manufacturer' not in ref_ds
+            else ref_ds.Manufacturer,
+            modality="" if 'Modality' not in ref_ds
+            else ref_ds.Modality,
+            patient_id=None if 'PatientID' not in ref_ds
+            else ref_ds.PatientID,
+            patient_name=None if 'PatientName' not in ref_ds
+            else ref_ds.PatientName,
+            patient_birth_date=None if 'PatientBirthDate' not in ref_ds
+            else ref_ds.PatientBirthDate,
+            patient_sex=None if 'PatientSex' not in ref_ds
+            else ref_ds.PatientSex,
+            accession_number=None if 'AccessionNumber' not in ref_ds
+            else ref_ds.AccessionNumber,
+            study_id=None if 'StudyID' not in ref_ds
+            else ref_ds.StudyID,
+            study_date=None if 'StudyDate' not in ref_ds
+            else ref_ds.StudyDate,
+            study_time=None if 'StudyTime' not in ref_ds
+            else ref_ds.StudyTime,
+            referring_physician_name=None if 'ReferringPhysicianName' not in
+            ref_ds else ref_ds.ReferringPhysicianName,
         )
-        _convert_legacy_to_enhanced(legacy_datasets, self)
-        self.PresentationLUTShape = 'IDENTITY'
+        self._legacy_datasets = legacy_datasets
+        self._perframe_functional_group = ـPerframeFunctionalGroup(
```

Comment:
I would use the plural for `_perframe_functional_groups` and `_shared_functional_groups`, since these represent sequences of functional groups